shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/impl/MySQLDMLVisitor.java

```diff
@@ -551,16 +551,21 @@ public ASTNode visitJoinSpecification(final JoinSpecificationContext ctx) {
         if (null != ctx.expr()) {
             ASTNode expr = visit(ctx.expr());
             if (expr instanceof PredicateSegment) {
-                PredicateSegment predicate = (PredicateSegment) expr;
-                result.setPredicateSegment(predicate);
+                result.setPredicateSegment((PredicateSegment) expr);
+                AndPredicate andPredicate = new AndPredicate();
+                andPredicate.getPredicates().add((PredicateSegment) expr);
```

Comment:
Why is the `expr` visited twice?

Reply:
In order to ensure that the previous integration test can pass, in the past, only tables with only one condition was taken, but it may be multiple conditions

Reply:
Not only test code,it's related to the business code, I can't change it

Reply:
If so, could we reomve `private PredicateSegment predicateSegment;`?